src/main/java/org/jvnet/hudson/plugins/repositoryconnector/VersionParameterDefinition.java

```diff
@@ -53,35 +58,74 @@ public VersionParameterDefinition(String repoid, String groupid,
     public VersionParameterDefinition copyWithDefaultValue(ParameterValue defaultValue) {
         if (defaultValue instanceof StringParameterValue) {
             // TODO: StringParameterValue value = (StringParameterValue) defaultValue;
-            return new VersionParameterDefinition(getRepoid(), "",
-                    "", "", getDescription());
+            return new VersionParameterDefinition(getRepoid(), getGroupid(), getArtifactid(), "", getDescription(),true);
         } else {
             return this;
         }
     }
 
+    public boolean reverseOrder() {
+    	return reverseOrder;
+    }
+    
+    public Collection<Repository> getRepos(String repoid) {
+    	List<Repository> out = new ArrayList<Repository>();
+    	
+    	for(Map.Entry<String,Repository> e : RepositoryConfiguration.get().getRepositoryMap().entrySet()) {
+    		if(repoid.compareTo("ALL")==0 || repoid.compareTo(e.getKey())==0) {
+    			out.add(e.getValue());
+    		}
+    	}
+    	return out;
+    }
+    
+    
     @Exported
     public List<String> getChoices() {
-        Repository r = DESCRIPTOR.getRepo(repoid);
-        List<String> versionStrings = new ArrayList<String>();
-        if (r != null) {
-            File localRepo = RepositoryConfiguration.get().getLocalRepoPath();
-            Aether aether = new Aether(DESCRIPTOR.getRepos(), localRepo);
-            try {
-                List<Version> versions = aether.resolveVersions(groupid, artifactid);
-                for (Version version : versions) {
-                    versionStrings.add(version.toString());
-                }
-            } catch (VersionRangeResolutionException ex) {
-                log.log(Level.SEVERE, "Could not determine versions", ex);
-            }
-            if (!versionStrings.isEmpty()) {
-                // reverseorder to have the latest versions on top of the list
-                Collections.reverse(versionStrings);
-                // add the default parameters
-                versionStrings.add(0, "LATEST");
-                versionStrings.add(0, "RELEASE");
+    	Collection<Repository> repos = getRepos(repoid);
+
+    	List<String> versionStrings = new ArrayList<String>();
+
+    	File localRepo = RepositoryConfiguration.get().getLocalRepoPath();
+        log.info("VersionParameterDefinition: local repo "+localRepo.getAbsolutePath());
+        
+        Aether aether = new Aether(
+        		repos, localRepo, null, false, 
+        		RepositoryPolicy.UPDATE_POLICY_ALWAYS, 
+                RepositoryPolicy.CHECKSUM_POLICY_FAIL, 
```

Comment:
yes, possible. however ... i don't think listing version (without downloading) would trigger checksum errors anyways ... what would it build a checksum on? also - i honestly can't remember when i had a checksum error the last time. i think we ran out of disk space? i don't know. also, maven would complain as well.  ... do you have that a lot? as in ... more than once a year? i really didn't think this would have been a relevant issue, but i could be wrong of course.